jobs/promptLicenceCreation.ts

```diff
@@ -110,21 +104,42 @@ const notifyComOfUpcomingReleases = async (emailGroups: EmailContact[]) => {
 }
 /* eslint-enable */
 
+const excludeCasesNotAssignedToPpWithinPast7Days = (caseload: ManagedCase[]): ManagedCase[] => {
+  const previousWeekStart = startOfISOWeek(subDays(new Date(), 7))
+  const previousWeekEnd = endOfISOWeek(subDays(new Date(), 7))
+
+  const isWithinPastWeek = (c: ManagedCase) => {
+    if (c.deliusRecord.allocationDate) {
+      const dateAllocatedToPp = new Date(c.deliusRecord.allocationDate)
+      return dateAllocatedToPp >= previousWeekStart && dateAllocatedToPp <= previousWeekEnd
+    }
+    return false
+  }
+
+  return caseload.filter(isWithinPastWeek)
+}
+
 Promise.all([
   pollPrisonersDueForLicence(
-    startOfWeek(add(new Date(), { weeks: 12 }), { weekStartsOn: 1 }),
-    endOfWeek(add(new Date(), { weeks: 12 }), { weekStartsOn: 1 }),
-    [LicenceStatus.NOT_STARTED],
-    'No licences that are NOT_STARTED'
+    startOfISOWeek(add(new Date(), { weeks: 12 })),
+    endOfISOWeek(add(new Date(), { weeks: 12 })),
+    [LicenceStatus.NOT_STARTED]
   ),
+
   pollPrisonersDueForLicence(
-    startOfWeek(new Date(), { weekStartsOn: 1 }),
-    endOfWeek(add(new Date(), { weeks: 3 }), { weekStartsOn: 1 }),
-    [LicenceStatus.NOT_STARTED, LicenceStatus.IN_PROGRESS],
-    'No licences that are NOT_STARTED or IN_PROGRESS'
-  ),
+    startOfISOWeek(add(new Date(), { weeks: 4 })),
+    endOfISOWeek(add(new Date(), { weeks: 11 })),
+    [LicenceStatus.NOT_STARTED]
+  ).then(caseload => excludeCasesNotAssignedToPpWithinPast7Days(caseload)),
+
+  pollPrisonersDueForLicence(startOfISOWeek(new Date()), endOfISOWeek(add(new Date(), { weeks: 3 })), [
```

Comment:
would be nice to rearrange this so it reads in natural order - week 1-3, 4-11 and then 12 rather than 12, 4-11, 1-3.
Maybe as part of the next PR?

Reply:
yep